src/persist-client/src/internal/compact.rs

```diff
@@ -915,7 +915,14 @@ where
                     };
                     timings.part_fetching += fetch_start.elapsed();
                     total_bytes += chunk.goodbytes();
-                    chunks.push(chunk.records().clone());
+                    chunks.push(
+                        chunk
+                            .get_or_make_codec::<K, V>(
+                                write_schemas.key.as_ref(),
+                                write_schemas.val.as_ref(),
+                            )
+                            .clone(),
+                    );
```

Comment:
Should we add a TODO here for consolidation on structured data?

Reply:
I think this one is captured by the existing todo... the other branch does preserve the structured data and should replace this branch entirely in production soon!